packages/core/src/state/merge/mergeDeepCopy.ts

```diff
@@ -5,15 +5,25 @@ import isMergeableObject from './isMergeable';
 /**
  * Deep merge two objects or arrays. Uses static merge function if exists.
  */
-export default function mergeDeepCopy<T1, T2>(target: T1, source: T2): T1 & T2 {
+export default function mergeDeepCopy<T1, T2>(
+  target: T1,
+  source: T2,
+  targetMeta?: any,
```

Comment:
couldn't this have a more useful type?
```suggestion
  targetMeta?: {
    data?: T2['date'],
    [k: string]: unknown,
  },
```

Reply:
that breaks in all usages in the function. this is recursively evaluated but since its not publicly exposed just making the typings simple seemed to work best. Furthermore, typescript disallows > comparisons for things possibly undefined, but we actually want to use the javascript behavior here to resolve that to false for fallbacks.

Reply:
gotcha. would `unknown` be better than `any`?

Reply:
unknown doesn't let me do ?. on it

Reply:
hm, it should - i guess TS types are broken for optional chaining.

Reply:
My theory is that they just think treating types as loosely as javascript does (where everything is an object) is not a good idea. They also do this for instance where I cannot do < comparison on undefined, even tho that has a valid result.

Reply:
I subscribed to that issue now :)